src/bin/addr2line.rs

```diff
@@ -56,16 +52,41 @@ fn main() {
             .multiple(true)
             .index(1))
         .after_help(POSTSCRIPT)
-        .get_matches();
+        .get_matches()
+}
+
+#[cfg(any(feature = "rustc-demangle", feature = "cpp_demangle"))]
+fn get_options(matches: &clap::ArgMatches) -> addr2line::Options {
+    let mut opts = addr2line::Options::default();
+
+    if matches.is_present("functions") {
+        opts = opts.with_functions();
+    }
+    if matches.is_present("demangle") {
+        opts = opts.with_demangling();
+    }
```

Comment:
This would be a lot more concise with `if cfg!`, though I'm unsure whether that would compile? In theory the compiler could be clever enough to let it work even when the features are off.